Riot/Modules/AuthenticatedSession/AuthenticatedSessionViewControllerFactory.swift

```diff
@@ -0,0 +1,216 @@
+/*
+ Copyright 2020 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import Foundation
+
+enum AuthenticatedSessionViewControllerFactoryErrorCode: Int {
+    case flowNotSupported = 0
+}
```

Comment:
Prefer use an `Error` conforming to `CustomNSError`. We can even conform to `LocalizedError` to return directly the localized description through `var errorDescription: String?` variable instead of using  `var errorUserInfo: [String: Any]`. 
After doing that you can remove `func unsupportedFlowError()` method.

```suggestion
enum AuthenticatedSessionViewControllerFactoryError: Int, Error, CustomNSError {
    case flowNotSupported = 0
    
    // MARK: - CustomNSError
    
    static let errorDomain = "AuthenticatedSessionViewControllerFactoryErrorDomain"
    
    var errorCode: Int { return self.rawValue }
    
    var errorUserInfo: [String: Any] {
        let userInfo: [String: Any]

        switch self {
        case .flowNotSupported:
            userInfo = [NSLocalizedDescriptionKey: VectorL10n.authenticatedSessionFlowNotSupported]
        }
        return userInfo
    }
}
```